.github/workflows/build.yml

```diff
@@ -40,36 +32,23 @@ jobs:
     runs-on: ubuntu-latest
     needs:
       - pre-commit
+    outputs:
+      version: ${{ steps.build-and-tag.outputs.version }}
     steps:
-      - uses: actions/checkout@v3
-
-      - uses: actions/cache@v3
-        with:
-          path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: |
-            ${{ runner.os }}-maven-
-
-      - name: Set up JDK
-        uses: actions/setup-java@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/maven-build-and-tag@v2.3.0
+        id: build-and-tag
         with:
-          java-version: '11'
-          distribution: 'temurin'
-
-      - name: Define Maven Command
-        id: define_maven_command
-        shell: bash
-        run: |
-          if [[ ${{ github.event_name }} == 'push' || ( ${{ github.ref_name }} == 'develop' && '${{ inputs.shouldDeploy }}' == 'true' ) ]]
-          then
-            echo "command=deploy" >> $GITHUB_OUTPUT
-          else
-            echo "command=verify" >> $GITHUB_OUTPUT
-          fi
-
-      - name: Build, Test and sometimes deploy
-        run: |
-          mvn ${{ steps.define_maven_command.outputs.command }} $MAVEN_CLI_OPTS
+          upload-jars: 'true'
+          upload-coverage: 'true'
```

Comment:
do we need to do so on PRs?

Reply:
Not sure if internally the action deals with PR situations differently, but this is the same approach we follow in other projects:

- https://github.com/Alfresco/alfresco-deployment-service/blob/develop/.github/workflows/build.yml#L54-L55
- https://github.com/Alfresco/alfresco-modeling-service/blob/develop/.github/workflows/build.yml#L32-L33
- https://github.com/Alfresco/alfresco-process/blob/develop/.github/workflows/build.yml#L33-L34

Reply:
Thanks, I was just wondering (and that's probably of low impact here, but upload of jars takes 5 to 10 mins on process and process-connector-services, so I wonder if we could skip this for PRs at least :grimacing:, not sure if we're confusing CI runs with nexus or if there's a real benefit, maybe this should not be done blindly everywhere)

Reply:
that is a valid concern
We do so in PRs from what I see here: https://github.com/Alfresco/alfresco-build-tools/blob/master/.github/actions/maven-build-and-tag/action.yml#L177-L183
it would be as easy as adding a check for a push event in the existing `if`

Reply:
Yes, I'll ask original commiter(s) if this can be refined :+1: